controllers/tournament.controller.js

```diff
@@ -0,0 +1,256 @@
+import { ObjectId } from "mongodb";
+import z, { ZodError } from "zod";
+import { clubMembers, tournaments } from "../config/db.js";
+
+const CREATE_SCHEMA = z.object({
+    name: z.string().trim().min(3).max(256),
+    clubId: z.string().trim().min(1),
+    startTime: z.string().optional(), //convert to datetime
+    settings: z.object({
+        rankingConfig: z.object({
+            winPoints: z.number().min(0).default(3),
+            drawPoints: z.number().min(0).default(1),
+            lossPoints: z.number().default(0), //set up a min value (can go negative)
+            addScorePoints: z.boolean().default(false),
+        }).optional(),
+    }).optional(),
+}).strict();
+
+/** @type {import("express").RequestHandler} */
+export async function createTournament(req, res) {
+    try {
+        const parsed = CREATE_SCHEMA.parse(req.body);
+
+        // Verify user is a member of the club
+        const membership = await clubMembers.findOne({
+            clubId: new ObjectId(parsed.clubId),
+            userId: new ObjectId(req.user.id),
+            role: { $in: ["owner"] },
+        });
+
+        if (membership == null) {
+            return res.status(403).json({
+                message: "You don't have permission to create tournaments in this club",
+            });
+        }
+
+        const { insertedId: tournamentId } = await tournaments.insertOne({
+            name: parsed.name,
+            clubId: new ObjectId(parsed.clubId),
+            created: new Date(),
```

Comment:
(I'll fix this later, but) Try to be more specific about time related property names from now on (in any project actually). `createdAt` speaks more about time than `created`, which feels more like a boolean value name. One should be able to easily tell which key is what (i.e., `createdAt` is time).